src/modules/auth/components/RoleDropdown.js

```diff
@@ -0,0 +1,173 @@
+import React, { useState, useEffect, useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Dropdown, DropdownToggle,
+	DropdownItem, DropdownMenu, Input,
+	UncontrolledTooltip } from "reactstrap";
+
+const RoleDropdown = React.memo(({props, tenantObj, selectedTenants, setSelectedTenants, idx}) => {
+
+	const { t } = useTranslation();
+	const [dropdownOpen, setDropdownOpen] = useState(false);
+	const [displayTenantRoles, setDisplayTenantRoles] = useState({data: []});
+	const [count, setCount] = useState(0);
+	const [filter, setFilter] = useState("");
+	const [limit, setLimit] = useState(5);
+
+	let SeaCatAuthAPI = props.app.axiosCreate('seacat_auth');
+
+	useEffect(() => {
+		retrieveRoleList(tenantObj._id ?? undefined);
+	}, [limit]);
+
+	// this useEffect should match selected roles with the ones we want to display and re-add the ones which were removed from selected roles in BulkAssignmentContainer
+	useEffect(() => {
+		if (tenantObj.selectedRole && (tenantObj?.selectedRole?.length > 0)) {
+			let display = {...displayTenantRoles};
+			// newDisplayData will hold roles coming from the service minus the ones we've already assigned
+			let newDisplayData = [];
+			tenantObj?.roles?.data.map((tenantRole) => {
+				// we are comparing already selected roles with all roles and keeping only the elements, which are not included in selectedRole(s)
+				let found = tenantObj?.selectedRole.find(el => el === tenantRole._id);
+				if (!found) {
+					newDisplayData.push(tenantRole);
+				}
+			});
+			display['data'] = newDisplayData;
+			setDisplayTenantRoles(display);
+		} else {
+			setDisplayTenantRoles(tenantObj.roles);
+		}
+		setCount(tenantObj?.roles?.count);
+	}, [selectedTenants]);
+
+	// TODO: uncomment, when search functionality in roles is enabled on the backend
+	// // sets 0.5s delay before triggering the search call when filtering through tennants
+	// useEffect(() => {
+	// 	if (timeoutRef.current !== null) {
+	// 		clearTimeout(timeoutRef.current);
+	// 	}
+	// 	timeoutRef.current = setTimeout(() => {
+	// 		timeoutRef.current = null;
+	// 		retrieveRoleList(tenantID);
+	// 	}, 500);
+	// }, [filter]);
+
+	// const handleFilter = (e) => {
+	// 	setLimit(5);
+	// 	setFilter(e.target.value);
+	// }
+
+	// fetch roles for Tenant dropdowns
+	const retrieveRoleList = async (tenantId) => {
+		let response;
+		let objCopy;
+		let id = tenantId;
+		let parameters = {f: filter, i: limit, exclude_global: true};
+		if (tenantId === undefined) {
+			id = '*';
+			parameters['exclude_global'] = false;
+			objCopy = {...tenantObj};
+		} else {
+			objCopy = selectedTenants.find(obj => obj._id === tenantId);
+		}
+		try {
+			response = await SeaCatAuthAPI.get(`/role/${id}`, {params: parameters});
+			let selectedTenantsCopy = [...selectedTenants];
+			let ten = {...objCopy, roles: response.data};
```

Comment:
@petrKavulok use a better naming of `ten` variable, I suppose its a `tenantsRolesObject` or something